src/core/dom/layout/viewport-observer.js

```diff
@@ -39,10 +39,27 @@ const viewportCallbacks = new WeakMap();
  *
  * @param {Element} element
  * @param {function(IntersectionObserverEntry):void} callback
+ * @param {import('./types').IOOptions} [opts]
  * @return {import('#core/types/function/types').UnlistenCallback} clean up closure to unobserve the element
  */
-export function observeIntersections(element, callback) {
+export function observeIntersections(element, callback, opts) {
   const win = getWin(element);
+
+  if (opts) {
```

Comment:
Agreed, the main purpose for this file is caching ... however, with a name like `observeIntersections`, caching is an implementation detail that is hidden from the consumer.  
We could create a `observeIntersectionsNoCache`, but that forces the consumer to figure out why or why not to use caching.
We could possibly create a `observeIntersectionsWithOptions` method, which doesn't perform the caching.  But I feel like  that's a great case for just using an optional `opts` parameter.

Reply:
hmm yeah i get what you're saying. I was thinking this logic should belong a function custom to amp-list, but this is called transitively so that gets messy too.